source/github-helpers/search-query.ts

```diff
@@ -95,6 +100,10 @@ export default class SearchQuery {
 		const parts = splitQueryString(query);
 		const cleaned = cleanQueryParts(parts);
 		this.searchParams.set('q', cleaned.join(' '));
+		if (this.link?.pathname.includes('/labels/')) {
+			// Avoid a redirection to the conversation list that would erase the search query #5176
+			this.link.pathname = this.link.pathname.replace(/\/labels\/.+$/, '/issues');
```

Comment:
SearchQuery is a mess, sorry about that. We try to parse, clean and update elements on creation. That’s a lot of side effects for a class. 

For this PR this line can be merged, but know that just calling this class on a /label/ link will automatically drop this /label/ before even attempting any changes. 

The idea was to make it easy to update existing links, but perhaps we should avoid changing the original link until .commit() or .updateOriginal() is called. Alternative suggestions welcome.